Adds `create_point` attribute to data collections

mockoocy opened the pull request:
This parameter is optional.
It's purpose is to allow generic tasks to create a 2d point when it's ran with no shape selected. Similarily to the Data Collection task. Creating the point will be relegated to frontend, as screen coordinates are required to create a 2D point.

Default value of `False` does not do anything.

This PR does not change any present behaviour (other than few more bytes in `/available_tasks` request :smiley_cat: ).
It will do something when [this frontend PR](https://github.com/mxcube/mxcubeweb/pull/1619) is merged :)

I am not really fond of how it's solved there, but I have no better idea  :shrug:

marcus-oscarsson commented on mxcubecore/model/common.py:
```diff
@@ -47,6 +47,8 @@ class StandardCollectionParameters(BaseModel):
     selection: list = Field([])
     shape: str = ""
+    # When set to True, frontend will create a point when starting task without a shape selected.
+    create_point: bool = False
```
So what you would like to do is to collect the current position without necessarily having a point before ? 

That could probably be part of your collect logic, `if no point: add ...` would that work ?

marcus-oscarsson replied:
For doing that you could look at `MiniDiff.save_current_position()`

mockoocy replied:
Not really, the thing that I want to achieve is to create a point at location where user right-clicked to open context menu.

In other words, I want to have something that works like so:
- user right clicks anywhere on the screen
- selects the custom task on the context menu
- a 2d point is created there
- the point is passed to the task as a parameter.

marcus-oscarsson replied:
Alright I see. I think that feature already exists, if I'm not mistaken. At least for 2D-points. Its very rarely used, but it used to work.

How is the `create_point` parameter meant to be used, I am not sure its needed. You can do what you describe above in anyways its just the the data collection option needs to be available in the context menu without having a point.

mockoocy replied:
At MAXIV we have some task, that always sets `create_point = True` in the parameters.
On the frontend, the `create_point` parameter is recognized, and if set to `true`, then first 2d point is created, then it's passed to the task (you can see it at the PR linked in the description of this PR - it's very similar to how it's done in normal Data Collection OSC task).

The thing is, there are two pieces of information that must we must know:
- should we create a point for the particular generic task
- where to create it (location of the point on the sample view)

So I created this attribute, to have information if a point should be created for a particular task or not.
And then frontend would pass the `screenCoords` of the point that has been clicked on the sample view.

The other way of doing it, I can think of is to pass the screen coordinates to the task parameters, when no point is selected. Then based on them, a point could be created 'inside' the task.

marcus-oscarsson replied:
It perhaps helps if we discuss this in quick meeting, I think I'm missing something. Perhaps what could help is to compare the functionality you would like with the current 2D point behavior and for instance the `Data Collection (Limited OSC)` collection

![image](https://github.com/user-attachments/assets/bc73e447-b360-4478-9a25-7f628deb7924)

marcus-oscarsson replied:
Does this feature already do something similar to what you need ?

mockoocy replied:
`Data Collection (Limited OSC)` indeed does something similar. I am down for some quick meeting to explain it further :)

marcus-oscarsson replied:
Cool :), How about tomorrow ?

mockoocy replied:
Okay, I'm free anywhere between 9:00 and 15:00

marcus-oscarsson replied:
Ok, Ill send a mail to the others as well should we say 9.30h preliminary then ?

mockoocy replied:
Sure

mockoocy replied:
As we discussed this morning, 
I removed the `create_point` attribute there.  Instead it's all done on the frontend.
Also I created an enum `TaskPrerequisites` to serve as an alternative to writing string literals in the `BaseQueueEntry.REQUIRES`;
i.e. `REQUIRES = ['point'] ` would be replaced by `REQUIRES = [TaskPrerequisites.POINT]`.
Added some docs for it too.

mockoocy committed:
Adds enum for QueueEntry.REQUIRES configuration

Previously custom queue entries had to specify "supported" shapes
in the `BaseQueueEntry.REQUIRES` class variable using list of names of
these shapes.

It makes it possible to specify these names with an enumeration instead
to make it more readable and better documented.

Also changes documentation to reflect on this change;
now in the code snippet describing creating custom queue entries, it
uses the new way of doing it.

It does not break old code; client application receives same string
values as before in the list

mockoocy committed:
Makes TestCollection use `TaskPrerequisite` enum

walesch-yan reviewed (approved):
Looks definitely better with the enums. Thanks!:)
